plugins/org.yakindu.sct.doc.user/src/user-guide/embedded_systems_integration_guide.textile

```diff
@@ -0,0 +1,281 @@
+
+h1(#embedded_systems_integration_guide). Embedded Systems Integration Guide
+
+State Machine code generated by YAKINDU Statechart Tools can effectively be used on Embedded Systems. The integration contains different parts, but can mostly be described in a generic way, as most steps of the integration are platform independent.
+The differences of the integration mostly stemp from the execution - CycleBased or EventDriven - of the state machine or the used functionalities, such as in, out and timed events. Other best practice implementation patterns can be based on whether a bare metal microcontroller or an operating system is used. So, there are different factors which may have a impact on the integration:
+
+In the statechart:
+* State Machine execution
+** CycleBased
+** EventDriven
+* Used event types
+** In events
+** Time events
+** Out events
+In the environment:
+* Bare-metal or RTOS
+* Usage of interrupts or polling
+* Usage of sleep modes
+
+This guide will give an overview of the most common and well known design patterns in a generic way and initially provides examples for an Arduino and only starts with the bare-metal integration. It will be continuously expanded.
+
+h2(#polling-vs-interrupts). Polling vs.Interrupts
+
+Programming bare-metal microcontrollers without any kind of RTOS or scheduler mainly leads to the architectural decision: Polling or Interrupts.
+This section will give a short overview, of how both design patterns work.
+
+h3(#polling). Polling
+
+Polling is a simple approach to get a first prototype running. Most programs look like this:
+
+bc.. 
+void main() {
+	init();
+	while(true){
+		readInputs();
+		doStuff();
+		updateOutputs();
+		delay_ms(100);
+	}
+}
+
+p. 
+The program starts with an initialization. After this, everything will be handled in a while loop, where the inputs will be polled cyclic - in this case very 100 milliseconds plus the execution time the program. Wating for x milliseconds must not be realized in a blocking delay function, it also can be done by using sleep modes and a timer interrupt. But the main idea of polling is cyclic reacting on events.
+For applications where events/inputs only rarely occur, this is not a good design choice, as the sensor are frequently read. This will lead to higher power consumption.
```

Comment:
```suggestion
For applications where events/inputs only rarely occur, this is not a good design choice, as the sensors are read frequently. This will lead to higher power consumption.
```